CliWrap/ICli.cs

```diff
@@ -99,5 +99,10 @@ public interface ICli
         /// Executes the process and doesn't wait for it to exit.
         /// </summary>
         void ExecuteAndForget();
+
+        /// <summary>
+        /// Wheny using ExecuteAsync, use this to immediatelly get the process id
+        /// </summary>
+        int ProcessId { get; }
```

Comment:
Let's use `int?` here because there's no technical guarantee a process cannot have ID of 0.
Also update tests to use `Assert.That(cli.ProcessId, Is.Not.Null);`

Reply:
And update summary to `Process ID associated with the last execution or null if the process hasn't been started yet.`